cli/lsp/documents.rs

```diff
@@ -1162,28 +1151,32 @@ impl Documents {
 
   pub fn update_config(
     &mut self,
+    root_dirs: Vec<Url>,
     maybe_import_map: Option<Arc<import_map::ImportMap>>,
     maybe_config_file: Option<&ConfigFile>,
     maybe_package_json: Option<&PackageJson>,
     npm_registry_api: NpmRegistryApi,
     npm_resolution: NpmResolution,
   ) {
     fn calculate_resolver_config_hash(
+      root_dirs: &[Url],
       maybe_import_map: Option<&import_map::ImportMap>,
       maybe_jsx_config: Option<&JsxImportSourceConfig>,
       maybe_package_json_deps: Option<&BTreeMap<String, NpmPackageReq>>,
     ) -> u64 {
       let mut hasher = FastInsecureHasher::default();
+      hasher.write_hashable(&{
+        // ensure these are sorted
+        let mut root_dirs = root_dirs.to_vec();
+        root_dirs.sort();
+        root_dirs
+      });
       if let Some(import_map) = maybe_import_map {
         hasher.write_str(&import_map.to_json());
         hasher.write_str(import_map.base_url().as_str());
       }
-      if let Some(jsx_config) = maybe_jsx_config {
-        hasher.write_hashable(&jsx_config);
-      }
-      if let Some(deps) = maybe_package_json_deps {
-        hasher.write_hashable(&deps);
-      }
+      hasher.write_hashable(&maybe_jsx_config);
```

Comment:
**question:** Is it okay to remove the if and go with unconditional writing?

Reply:
Yup because `Option<Hashable>` implements `Hashable`.